src/main/java/io/nats/client/NatsThread.java

```diff
@@ -33,7 +29,7 @@ public NatsThread(Runnable runnable, String name) {
         super(runnable, name + "-" + created.incrementAndGet());
         setUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler() {
             public void uncaughtException(Thread t, Throwable e) {
-                logger.debug("UNCAUGHT in thread {}", t.getName(), e);
+               e.printStackTrace();
```

Comment:
Am I missing something here? Looks like the whole reason for this class is so that the API knows how many were created. To address @tylertreat comment, likely we just want to make use an executor, and just run the Runnables. It could be that this value could be tweaked to get more workers processing the runnables. Not sure what the intention was.